lib/model/shared_postcard.dart

```diff
@@ -4,19 +4,54 @@
 //  Use of this source code is governed by the BSD-2-Clause Plus Patent License
 //  that can be found in the LICENSE file.
 //
+import 'package:autonomy_flutter/util/constants.dart';
 import 'package:json_annotation/json_annotation.dart';
 
-part 'shared_postcard.g.dart';
-
 @JsonSerializable()
 class SharedPostcard {
   final String tokenID;
   final String owner;
+  final DateTime? sharedAt;
+
+  SharedPostcard(this.tokenID, this.owner, this.sharedAt);
+
+  // override operator ==
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is SharedPostcard &&
+          runtimeType == other.runtimeType &&
+          tokenID == other.tokenID &&
+          owner == other.owner;
+
+  // fromJson method
+  factory SharedPostcard.fromJson(Map<String, dynamic> json) {
+    return SharedPostcard(
+      json["tokenID"] as String,
+      json["owner"] as String,
+      json["sharedAt"] == null
+          ? null
+          : DateTime.parse(json["sharedAt"] as String),
+    );
+  }
 
-  SharedPostcard(this.tokenID, this.owner);
+  Map<String, dynamic> toJson() {
+    return {
+      "tokenID": tokenID,
+      "owner": owner,
+      "sharedAt": sharedAt?.toIso8601String(),
+    };
+  }
 
-  factory SharedPostcard.fromJson(Map<String, dynamic> json) =>
-      _$SharedPostcardFromJson(json);
+  bool get isExpired {
+    if (sharedAt == null) {
+      return false;
+    }
+    return DateTime.now()
+        .subtract(POSTCARD_SHARE_LINK_VALID_DURATION)
+        .isAfter(sharedAt!);
+  }
 
-  Map<String, dynamic> toJson() => _$SharedPostcardToJson(this);
+  @override
```

Comment:
if you want to the new share override the previous share, I think you should keep
 ```
int get hashCode => tokenID.hashCode ^ owner.hashCode;
```